src/audio/module_adapter/iadk/system_agent.cpp

```diff
@@ -118,17 +118,15 @@ int SystemAgent::CheckIn(ProcessingModuleFactoryInterface& module_factory,
 } /* namespace system */
 } /* namespace intel_adsp */
 
-#ifdef __cplusplus
-extern "C" {
-#endif
 /* The create_instance_f is a function call type known in IADK module. The module entry_point
```

Comment:
I'm a bit puzzled by removal of extern "C". On one hand I guess it can work as the function doesn't really use any C++ constructs, so the linkage is probably anyways the same. But why does the header still have the extern "C" declaration for system_agent_start(). Shouldn't the two match (either both can be removed or both need to be kept).

Reply:
I guess the `#ifdef` can be removed, because this **is** a C++ file, so you could just keep `extern "C" {`

Reply:
`extern "C" {}` with no `#ifdef`?

Reply:
@lyakh In the header file, the `system_agent_start` function is marked as `extern C` and there is no need to duplicate this in the cpp file.

Reply:
@lyakh @softwarecki  This should be ok. For most compilers if the linkage classification is different at declaration and definition, the first definition will take priority and in many codebases this is recommended convention.